core-api/src/main/java/com/optimizely/ab/config/audience/UserAttribute.java

```diff
@@ -20,27 +20,33 @@
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.optimizely.ab.config.ProjectConfig;
-import com.optimizely.ab.config.audience.match.Match;
-import com.optimizely.ab.config.audience.match.MatchType;
-import com.optimizely.ab.config.audience.match.UnexpectedValueTypeException;
-import com.optimizely.ab.config.audience.match.UnknownMatchTypeException;
+import com.optimizely.ab.internal.ConditionUtils;
+import com.optimizely.ab.internal.InvalidAudienceCondition;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import javax.annotation.concurrent.Immutable;
-import java.util.Collections;
 import java.util.Map;
 
 /**
- * Represents a user attribute instance within an audience's conditions.
+ * Represents a leaf node in audience condition AST.
+ *
+ * Defines a matching operation to evaluate on a specific attribute.
+ * The behavior of {@link #evaluate(ProjectConfig, Map)} is determined by specified
+ * match operation.
+ *
+ * If the given match type and value are not compatible, i.e. a substring match with non-string value,
+ * this class will always evaluate to UNKNOWN by returning {@code null}.
+ *
+ * TODO(llinn) convert this to a POJO and separate from {@link Condition} evaluation
```

Comment:
Not following POJO in this context. Do you mean, convert it to be encapsulated by a leaf node rather than directly acting as the leaf node? Or something like that?